File: VB/WinWebSolution.Module/MyEvent.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace WinWebSolution.Module
	'[DefaultClassOptions]
	Public Class CustomReferenceTypeField
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _Name, value)
			End Set
		End Property
	End Class
	<DefaultClassOptions> _
	Public Class MyEvent
		Inherits DevExpress.Persistent.BaseImpl.Event
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _CustomReferenceTypeField As CustomReferenceTypeField
		<RuleRequiredField(Nothing, DefaultContexts.Save)> _
		Public Property CustomReferenceTypeField() As CustomReferenceTypeField
			Get
				Return _CustomReferenceTypeField
			End Get
			Set(ByVal value As CustomReferenceTypeField)
				SetPropertyValue("CustomReferenceTypeField", _CustomReferenceTypeField, value)
			End Set
		End Property
		Private _CustomSimpleTypeField As String
		<RuleRequiredField(Nothing, DefaultContexts.Save)> _
		Public Property CustomSimpleTypeField() As String
			Get
				Return _CustomSimpleTypeField
			End Get
			Set(ByVal value As String)
				SetPropertyValue("CustomSimpleTypeField", _CustomSimpleTypeField, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/WinWebSolution.Module/MyEvent.cs
using System;
using DevExpress.Xpo;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace WinWebSolution.Module {
    //[DefaultClassOptions]
    public class CustomReferenceTypeField : BaseObject {
        public CustomReferenceTypeField(Session session) : base(session) { }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
    }
    [DefaultClassOptions]
    public class MyEvent : DevExpress.Persistent.BaseImpl.Event {
        public MyEvent(Session session) : base(session) { }
        private CustomReferenceTypeField _CustomReferenceTypeField;
        [RuleRequiredField(null, DefaultContexts.Save)]
        public CustomReferenceTypeField CustomReferenceTypeField {
            get { return _CustomReferenceTypeField; }
            set { SetPropertyValue("CustomReferenceTypeField", ref _CustomReferenceTypeField, value); }
        }
        private string _CustomSimpleTypeField;
        [RuleRequiredField(null, DefaultContexts.Save)]
        public string CustomSimpleTypeField {
            get { return _CustomSimpleTypeField; }
            set { SetPropertyValue("CustomSimpleTypeField", ref _CustomSimpleTypeField, value); }
        }
    }
}